### DISCUSSION TITLE
Problems when create ports dynamically on connect event

### DISCUSSION THREAD

**Discussion by Biowavix** [clientIO/joint#2011] [Q&A] [answered]
Hello everyone, I have a problem trying to add ports dynamically on link:connect event. I define a component with one "generic" input port. The outputs connected to this port, creates a new "generic" input port inside link:connect event. With this, you can create inputs ports directly by adding connections. 
The problem is that when I connect one output to this generic input port, the new port is created correctly and makes the input ports change their positions to fit this new port. This causes the link to no longer point to the port correctly, but to point to the position the port was in before moving. How can I make the link continue pointing to the port correctly?

I create a codepen with a very basic example to see the problem:
https://codepen.io/djimenez-developair/pen/WNKJYPX

Thank you in advance!

Edit: Note that if you move the component after the connection, the link target is fixed automatically

**Comment by kumilingus** [comment, marked as answer]
Hi, it's a bug. If you change `paper.options.async` to `false` the problem disappears.

As a workaround, you can add the port synchronously (see [here](https://codepen.io/jointjs/pen/Yzjvzmy/c1e605a2c3dd37911a3b7d52e04c401f)).
```js
this.addPort(port, { async: false });
```
I'll convert this into an issue. Thank you!

**Reply by kumilingus** [reply]
The issue: https://github.com/clientIO/joint/issues/2012

**Reply by Biowavix** [reply]
Thanks for the answer! The fix works correctly, I'll keep an eye on the issue.
